### DISCUSSION TITLE
how to delete port_forward

### DISCUSSION THREAD

**Discussion by yuecheng88** [jaredhendrickson13/pfsense-api#435] [Q&A]
I need to create some nat rule to let my internal machine can be accessible, also need delete it .But port_forward without list api ,so I cannot find this id【if i have delete ,the id can change】.

so the bad plan is :
1. get  list to find the new id is【when create i will use descr for only. 】
2. delete with new id.

I hope have this api for delete use the only identification.  in this api you can first find the id by the only identification,and then delete.

**Comment by jaredhendrickson13** [comment]
You can use the GET /api/v1/firewall/nat/port_forward endpoint to list all existing port forwards. You can obtain the IDs of port forwards as described in the 'Obtaining object IDs' section of the [Queries section in README.md](https://github.com/jaredhendrickson13/pfsense-api#queries). 

Once you have the ID of the object, you can delete with a DELETE request to /api/v1/firewall/nat/port_foward?id=ADD_ID_HERE. 

Since pfSense uses dynamic IDs, you will need to make another GET request to /api/v1/firewall/nat/port_forward to get the updated IDs of port forwards before making more DELETE requests as their IDs may have changed.
